### PR TITLE
Business Share Revenue

### PR DESCRIPTION
**Pull request by clostao**
## Description

A fee system where dApps will receive a share of the gas used in their contracts. 

Bug fixed in `fp_rpc::EthereumRuntimeRPCApi<Block>::account_basic` where the source for the `balanceOf` address would be a smart contract when its balance is read. 

## Types of changes

What types of changes does your code introduce?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Review by deblanco — CHANGES_REQUESTED** [review]
Missing docs.

**Comment by GabrielMartinezRodriguez** [comment]
> Missing docs.

**Comment by GabrielMartinezRodriguez** [comment]
> Missing docs.

done

File: primitives/runner/src/lib.rs [hunk 1]
```diff
@@ -224,12 +231,34 @@ where
...
+			FC::pay_fees(token, conversion_rate, actual_fee, validator, dapp).map_err(|_| {
+				RunnerError {
+					error: Error::<T>::FeeOverflow,
+					weight,
+				}
+			})?;
+
+		executor
+			.log(
+				H160::zero(),
```
Comment (GabrielMartinezRodriguez) [inline primitives/runner/src/lib.rs:244]: Maybe is good idea use the `FeeRewardsVaultController` as from instead of `0`

Comment (clostao) [reply primitives/runner/src/lib.rs]: Already fixed

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -789,7 +797,25 @@ impl fp_self_contained::SelfContainedCall for RuntimeCall {
 		len: usize,
 	) -> Option<TransactionValidity> {
 		match self {
-			RuntimeCall::Ethereum(call) => call.validate_self_contained(info, dispatch_info, len),
+			// IMPORTANT: TO REMOVE BEFORE MERGING
```
[the remark this answers is not in the document]
Comment (clostao) [reply runtime/src/lib.rs:800]: true

[the remark this answers is not in the document]
Comment (clostao) [reply runtime/src/lib.rs]: As we discussed, we're merging it and replacing it when custom pallet_balances is implemented

**Review by deblanco — CHANGES_REQUESTED** [review]

**Review by deblanco — APPROVED** [review]

**Review by GabrielMartinezRodriguez — APPROVED** [review]

**Pull request merged**